solutions/standard/main.tf

```diff
@@ -0,0 +1,336 @@
+  depends_on      = [ibm_iam_authorization_policy.kms_policy]
+  create_duration = "30s"
+}
+
+# Create auth policy (scoped to exact KMS key for backups)
+resource "ibm_iam_authorization_policy" "backup_kms_policy" {
+  count                    = local.create_cross_account_backup_kms_auth_policy ? 1 : 0
+  provider                 = ibm.kms
+  source_service_account   = local.account_id
+  source_service_name      = "databases-for-mysql"
+  source_resource_group_id = module.resource_group.resource_group_id
+  roles                    = ["Reader"]
+  description              = "Allow all MySQL instances in the resource group ${module.resource_group.resource_group_id} in the account ${local.account_id} to read the ${local.backup_kms_service} key ${local.backup_kms_key_id} from the instance GUID ${local.backup_kms_instance_guid}"
+  resource_attributes {
+    name     = "serviceName"
+    operator = "stringEquals"
+    value    = local.backup_kms_service
+  }
+  resource_attributes {
+    name     = "accountId"
+    operator = "stringEquals"
+    value    = local.backup_kms_account_id
+  }
+  resource_attributes {
+    name     = "serviceInstance"
+    operator = "stringEquals"
+    value    = local.backup_kms_instance_guid
+  }
+  resource_attributes {
+    name     = "resourceType"
+    operator = "stringEquals"
+    value    = "key"
+  }
+  resource_attributes {
+    name     = "resource"
+    operator = "stringEquals"
+    value    = local.backup_kms_key_id
+  }
+  # Scope of policy now includes the key, so ensure to create new policy before
+  # destroying old one to prevent any disruption to every day services.
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_backup_kms_authorization_policy" {
+  count           = local.create_cross_account_backup_kms_auth_policy ? 1 : 0
+  depends_on      = [ibm_iam_authorization_policy.backup_kms_policy]
+  create_duration = "30s"
+}
+
+#######################################################################################################################
+# MySQL admin password
+#######################################################################################################################
+
+resource "random_password" "admin_password" {
+  count            = var.admin_pass == null ? 1 : 0
+  length           = 32
+  special          = true
+  override_special = "-_"
+  min_numeric      = 1
+}
+
+locals {
+  # _- are invalid first characters
+  # if - replace first char with J
+  # elseif _ replace first char with K
+  # else use asis
+  generated_admin_password = startswith(random_password.admin_password[0].result, "-") ? "J${substr(random_password.admin_password[0].result, 1, -1)}" : startswith(random_password.admin_password[0].result, "_") ? "K${substr(random_password.admin_password[0].result, 1, -1)}" : random_password.admin_password[0].result
```

Comment:
There is a bug in other ICD DAs which needs to be fixed here too: https://github.ibm.com/GoldenEye/issues/issues/12231